src/event/cloudevent.ts

```diff
@@ -0,0 +1,85 @@
+const { v4: uuidv4 } = require("uuid");
+
+import { CloudEventV1, validate as validateV1, CloudEventV1Attributes } from "./v1";
+import { CloudEventV03, validate as validateV03, CloudEventV03Attributes } from "./v03";
+import Extensions from "./extensions";
+
+export const enum Version { V1 = "1.0", V03 = "0.3" };
+
+/**
+ * A CloudEvent describes event data in common formats to provide
+ * interoperability across services, platforms and systems.
+ * @see https://github.com/cloudevents/spec/blob/v1.0/spec.md
+ */
+export class CloudEvent implements CloudEventV1, CloudEventV03 {
+  id: string;
+  type: string;
+  source: string;
+  specversion: Version;
+  dataContentType?: string;
+  dataSchema?: string; // TODO: Is a URI type needed?
+  subject?: string;
+  time?: string|Date;
```

Comment:
If we're converting to strings below, we should just use `string`.
If we're allowing `Date`, we should not convert to ISO string below.

```ts
    } else if (this.time instanceof Date) {
      this.time = this.time.toISOString();
    }
```

Reply:
The `time` property, per the [spec MUST adhere to RFC3339](https://github.com/cloudevents/spec/blob/v1.0/spec.md#time). This enforces that.